pkg/cloud/services/ec2/launchtemplate_test.go

```diff
@@ -82,10 +87,18 @@ func TestGetLaunchTemplate(t *testing.T) {
 		name               string
 		launchTemplateName string
 		expect             func(m *mock_ec2iface.MockEC2APIMockRecorder)
-		check              func(launchtemplate *expinfrav1.AWSLaunchTemplate, userdatahash string, err error)
+		check              func(g *WithT, launchTemplate *expinfrav1.AWSLaunchTemplate, userDataHash string, err error)
 	}{
 		{
-			name:               "does not exist",
+			name: "Should not return launch template if empty launch template name passed",
+			check: func(g *WithT, launchTemplate *expinfrav1.AWSLaunchTemplate, userDataHash string, err error) {
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(userDataHash).Should(BeEmpty())
+				g.Expect(launchTemplate).Should(BeNil())
```

Comment:
Can we extract this also to a function as it is used in multiple places?

Reply:
Yes I could do that if we are doing same check in all the test cases but in last 2 test case I am doing different check. So that's why I thought to put it in a separate `check field`.
```
g.Expect(err).NotTo(HaveOccurred())
g.Expect(userDataHash).Should(Equal(testUserDataHash))
g.Expect(launchTemplate).Should(Equal(wantLT))
```

Reply:
Yes I could do that if we are doing same check in all the test cases but in last 2 test case I am doing different check. So that's why I thought to put it in a separate check field.
```
g.Expect(err).NotTo(HaveOccurred())
g.Expect(userDataHash).Should(Equal(testUserDataHash))
g.Expect(launchTemplate).Should(Equal(wantLT))
```